src/charm.py

```diff
@@ -14,74 +16,72 @@
     WaitingStatus,
 )
 from ops.framework import StoredState
+from string import ascii_letters, digits
 
 logger = logging.getLogger(__name__)
 PEER = "mysql"
 
 
 class MySQLCharm(CharmBase):
     """Charm to run MySQL on Kubernetes."""
+
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.image = OCIImageResource(self, 'mysql-image')
+        self._stored.set_default(mysql_setup={})
+        password = "".join(
+            random.choice(ascii_letters + digits) for x in range(20)
+        )
+        self._stored.set_default(MYSQL_ROOT_PASSWORD=password)
+        self.image = OCIImageResource(self, "mysql-image")
         self.framework.observe(self.on.start, self._on_start)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
 
     @property
-    def cluster_size(self) -> int:
-        """Return the size of the cluster."""
-        rel = self.model.get_relation(PEER)
-        return len(rel.units) + 1 if rel is not None else 1
+    def hostname(self) -> str:
+        """Return the Kubernetes hostname."""
+        unit_number = self._get_unit_number_from_unit_name(self.unit.name)
```

Comment:
If I am not mistaken you may be able to get the unit number from something like `self.unit.id` .

Reply:
AFAIK there is no `id` property in [ops.model.Unit](https://ops.readthedocs.io/en/latest/index.html?#ops.model.Unit) or should I check somewhere else?

Reply:
I could be wrong. I will dig when I get some time. If I find it will let you know